### PR TITLE
[8.11.x] BXMSPROD-1542 definition-file eval expression

### PR DESCRIPTION
**Pull request by Ginxo**
### JIRA
https://issues.redhat.com/browse/BXMSPROD-1542

### Referenced pull requests

- https://github.com/kiegroup/optaweb-vehicle-routing/pull/603
- https://github.com/kiegroup/optaweb-vehicle-routing/pull/602

I will propagate the change to drools, droolsjbpm-knowledge, and opta* projects as soon as we agree on the change

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>specific pull request build</b> please add comment: <b>Jenkins (re)run [optaweb-vehicle-routing] tests</b>
* for a <b>full downstream build</b> please add comment: <b>Jenkins run fdb</b>
* for a <b>compile downstream build</b> please add comment: <b>Jenkins run cdb</b>
* for a <b>full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>
* for an <b>upstream build</b> please add comment: <b>Jenkins run upstream</b>
* for a <b>Quarkus LTS check</b> please add comment: <b>Jenkins run LTS</b>
* for a <b>specific Quarkus LTS check</b> please add comment: <b>Jenkins (re)run [optaweb-vehicle-routing] LTS</b>
* for a <b>Native check</b> please add comment: <b>Jenkins run native</b>
* for a <b>specific Native LTS check</b> please add comment: <b>Jenkins (re)run [optaweb-vehicle-routing] native</b>
</details>

### CODE REVIEW

File: .github/workflows/pull_request.yml [hunk 1]
```diff
@@ -5,46 +5,35 @@ on: [pull_request]
...
         os: [ubuntu-latest, windows-latest]
-        java-version: [11]
+        java-version: [11, 17]
         maven-version: ['3.8.1']
       fail-fast: false
     runs-on: ${{ matrix.os }}
     name: Maven Build
     steps:
-      - name: Support longpaths
+      - name: Clean Disk Space
```
Comment (radtriste) [inline .github/workflows/pull_request.yml:19]: I would not change that part to be in sync with other 8.11.x/1.11.x branches

**Review by radtriste — APPROVED** [review]

File: .github/workflows/pull_request.yml [hunk 2]
```diff
@@ -40,11 +40,12 @@ jobs:
           restore-keys: ${{ runner.os }}-${{ matrix.java-version }}-maven${{ matrix.maven-version }}-m2
       - name: Build Chain ${{ matrix.java-version }}. Maven ${{ matrix.maven-version }}
         id: build-chain
-        uses: kiegroup/github-action-build-chain@v2.6.4
+        uses: kiegroup/github-action-build-chain@v2.6.18
         with:
           # OptaWeb has a different branching model than the rest of KIE. Every OptaWeb 8.x branch maps to main branch
           # in KIE, therefore "main" is hard-coded in the URL below. Once the branching model is unified and there is
```
Comment (dupliaka) [inline .github/workflows/pull_request.yml:46]: Shouldn't we remove this comment then?

Comment (Ginxo) [reply .github/workflows/pull_request.yml]: you are right @dupliaka thanks.
Comment updated

**Review by dupliaka — APPROVED** [review]
Thanks!

**Pull request merged**